src/com/ms/silverking/cloud/zookeeper/ZooKeeperExtended.java

```diff
@@ -42,1024 +43,1081 @@
+    try {
+      byte[] data;
+
+      data = getData(path, watcher, stat);
+      return data;
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
+  }
+
+  public String getString(String path) throws KeeperException {
+    byte[] data;
+
+    data = getByteArray(path);
+    return new String(data);
+  }
+
+  public String getString(String path, Watcher watcher, Stat stat) throws KeeperException {
+    byte[] data;
+
+    data = getByteArray(path, watcher, stat);
+    return new String(data);
+  }
+
+  public int getInteger(String path) throws KeeperException {
+    byte[] data;
+
+    data = getByteArray(path);
+    return NumConversion.bytesToInt(data);
+  }
 
-    public ChildrenCallbackOp getChildrenAsync(String path) throws KeeperException {
-        return getChildrenAsync(path, false);
+  public boolean exists(String path) throws KeeperException {
+    try {
+      return super.exists(path, false) != null;
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
-    
-    public void close() {
-        try {
-            super.close();
-        } catch (InterruptedException ie) {
-            throw new RuntimeException("panic");
-        }
+  }
+
+  public List<String> getChildren(String path, Watcher watcher) throws KeeperException {
+    try {
+      return super.getChildren(path, watcher);
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
-    
-    public static ZooKeeperExtended getZooKeeperWithRetries(ZooKeeperConfig zkConfig, int sessionTimeout, 
-                                                       Watcher watcher, int connectAttempts) 
-                                                       throws KeeperException, IOException {
-        ZooKeeperExtended   _zk;
-        int                 curAttempt;
-        
-        _zk = null;
-        curAttempt = 1;
-        do {
-            try {
-                _zk = new ZooKeeperExtended(zkConfig, sessionTimeout, watcher);
-                while (_zk.getState() == States.CONNECTING) {
-                    ThreadUtil.sleep(connectionCheckIntervalMillis);
-                }
-                break;
-            } catch (IOException ioe) {
-                Log.logErrorWarning(ioe);
-                if (curAttempt >= connectAttempts) {
-                    throw ioe;
-                }
-            } catch (KeeperException ke) {
-                Log.logErrorWarning(ke);
-                if (curAttempt >= connectAttempts) {
-                    throw ke;
-                }
-            }
-            curAttempt++;
-            ThreadUtil.randomSleep(0, (1 << curAttempt) * 1000);
-        } while (curAttempt < connectAttempts);
-        return _zk;
+  }
+
+  public List<String> getChildren(String path) throws KeeperException {
+    try {
+      return getChildren(path, false);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)